Libraries/Utilities/defineLazyObjectProperty.js

```diff
@@ -35,7 +35,21 @@ function defineLazyObjectProperty<T>(
     // `setValue` which calls `Object.defineProperty` which somehow triggers
     // `getValue` again. Adding `valueSet` breaks this loop.
     if (!valueSet) {
-      setValue(get());
+      try {
+        // Set `valueSet` to true optimistically to not fail into another infinite
+        // loop when property getter would try to access already defined property.
+        // This is specifically the case with 'regenerator-runtime' module, which
+        // checks if `global.regeneratorRuntime` property is already defined and
+        // thus triggers `getValue()` again.
+        valueSet = true;
```

Comment:
So basically you suggest fixing this in `regenerator`? I'm fine with that, but i think this rings a bell. Honestly, I don't know whether it should be expected or not but I'd say that we should do something to stop recursion, because nobody can guarantee that at some moment we won't run into the same situation again.